packages/expo-cli/src/commands/eas-build/build/action.ts

```diff
@@ -165,6 +166,17 @@ async function startBuildAsync<T extends Platform>(
     }
     if (!builder.ctx.commandCtx.skipProjectConfiguration) {
       try {
+        await setUpdateVersionsAsync({
+          projectDir: builder.ctx.commandCtx.projectDir,
+          nonInteractive: builder.ctx.commandCtx.nonInteractive,
+        });
+
+        if (!(await isUpdatesConfiguredAsync(builder.ctx.commandCtx.projectDir))) {
+          throw new Error(
+            '"expo-updates" is installed in the project, but the configuration is not up-to-date. Please run "expo eas:build:init" first to configure the project'
+          );
+        }
```

Comment:
```suggestion
        if (!(await isUpdatesConfiguredAsync(builder.ctx.commandCtx.projectDir))) {
          throw new Error(
            '"expo-updates" is installed in the project, but the configuration is not up-to-date. Please run "expo eas:build:init" first to configure the project'
          );
        }
        await setUpdateVersionsAsync({
          projectDir: builder.ctx.commandCtx.projectDir,
          nonInteractive: builder.ctx.commandCtx.nonInteractive,
        });

```
don't check in `isUpdatesConfiguredAsync` if versions are set correctly (just if they are)